### PR TITLE
async output parser

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

**Comment by nfcampos** [comment]
@hwchase17 Added ainvoke here (in case a subclass actually implements aparse we should call it directly. LGTM

**Review by nfcampos — APPROVED** [review]

File: libs/langchain/langchain/schema/output_parser.py [hunk 1]
```diff
@@ -27,6 +28,20 @@ def parse_result(self, result: List[Generation]) -> T:
...
+        """Parse a list of candidate model Generations into a specific format.
+
+        Args:
+            result: A list of Generations to be parsed. The Generations are assumed
+                to be different candidate outputs for a single model input.
+
+        Returns:
+            Structured output.
+        """
+        return await asyncio.get_running_loop().run_in_executor(
```
Comment (shroominic) [inline libs/langchain/langchain/schema/output_parser.py:41]: Shouldn't this also be an abstract method?

Comment (shroominic) [reply libs/langchain/langchain/schema/output_parser.py:41]: Ahh okay I think i understand now its running in a thread and resolving the blocking io

Comment (nfcampos) [reply libs/langchain/langchain/schema/output_parser.py:41]: Yes, and subclasses that actually have a real async implementation can override this method

**Pull request merged**